### DISCUSSION TITLE
Single step optimisation

### DISCUSSION THREAD

**Discussion by MarcFacerias** [casadi/casadi#3198] [Q&A] [answered]
Hello, 

I was looking into using casadi within a distributed optimization stack and I need to perform a single step of the optimization problem before sharing the variables within the system. Right now I am bypassing the error that the solver throws when reaching maximum iterations (in this case 1) and rebuilding the OP from scratch. I'm afraid this could lead to prohibitive computational times related to rebuilding the problem and the error handling embedded in the solver, in this case IPOPT. I was wondering if there's any recommendation regarding updating part of the OP and doing that single step in a faster manner. 

Thanks a lot!

**Comment by jgillis** [comment, marked as answer]
CasADi allows parametric optimization problems. If you can cast the needed modifications in your scheme in terms of parameters, you may skip the construction of the problem entirely, in stead simple call the existing solver with different numerical values for the 'p' argument.
